src/Predavanje02/Inventory.kt

```diff
@@ -1,65 +1,80 @@
 package Predavanje02
 
+import kotlin.system.exitProcess
 
 
 fun main() {
     println("Inventory management system")
     val inventory = Inventory()
+    val loginHelper = LoginHelper()
     var exit = 0
     println("Login name:")
     val username = readln()
-    if (login(Employee.employees, username, inventory)) {
-        do {
-            println(inventory.currentUser.name)
-            println("Press number: 1. Add Employee, 2. Add Product, 3. Update Stock, 4. Print all products 5. Exit")
-            when (readln().toInt()) {
-                1 -> {
-                    println("Name: ")
-                    val name = readln()
-                    println("Last name: ")
-                    val lastName = readln()
-                    println("Add role: 1. Admin, 2. Manager, 3. Worker")
-                    val roleValue = readln().toInt()
-                    val role = Role.entries.find { it.value == roleValue }
-                    inventory.currentUser.addEmployee(Employee(name, lastName, role!!))
-                }
-                2 -> {
-                    println("Title: ")
-                    val title = readln()
-                    println("Category: 1. Food, 2. Tech, 3. Book")
-                    val category = readln().toInt()
-                    val productCategory = ProductCategory.entries.find { it.value == category }
-                    println("Quantity: ")
-                    val quantity = readln().toInt()
-                    inventory.addToInventory(Product(title, productCategory!!, quantity))
+    loginHelper.performLogin(
+        username = username,
+        onFailure = { exitProcess(0) }) { currentUser ->
+        // Proceed with the program. The current user is available here if needed.
+    }
+    do {
+        println(inventory.currentUser.name)
+        println("Press number: 1. Add Employee, 2. Add Product, 3. Update Stock, 4. Print all products 5. Exit")
+        when (readln().toInt()) {
+            1 -> {
+                println("Name: ")
+                val name = readln()
+                println("Last name: ")
+                val lastName = readln()
+                println(buildString {
+                    Role.entries.forEachIndexed { index, role -> append("${index + 1}.$role ") }
```

Comment:
Na ovaj način nećeš morati mijenjati kod ako budeš dodao još neke role u program.